lib/as2.js

```diff
@@ -0,0 +1,71 @@
+// as2.js
+//
+// conversion routine for AS1 -> AS2
+//
+// Copyright 2017 AJ Jordan <alex@strugee.net>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+"use strict";
+
+var _ = require("lodash");
+
+
+var toAS2 = function(obj) {
+    var as2 = _.cloneDeep(obj);
+
+    as2["@id"] = as2.id;
+    delete as2.id;
+
+    if (as2.verb === "post") {
```

Comment:
If you change the verb, need to change this [#L145](https://github.com/pump-io/pump.io/blob/master/lib/model/activity.js#L145) or change the name of `Activity.prototype.apply[Method]`

Reply:
Don't need to change anything outside of this file since this runs so late in the request-response cycle. Basically everywhere else doesn't have to care.

Reply:
You only change the output/response?

What will happened with the petitions?, if comes the request with AS2, it transform to AS1 before process/save?

Reply:
@vxcamiloxv we don't support AS2 submissions yet but when we do that'll be how it works. At least for the time being.

Eventually we'll want to move to AS2 storage but. Not yet. We want the API to stabilize, etc.

Reply:
I think will be strange (and confusing), if the input is AS1 and the output AS2, for the external apps could be more work (migrate the output and later the input) and probability cause more side effects and more errors.

I think is better one big change a not change by parts, is better one migration than 2 big migrations in a short period of time, mostly when is related with the input/output of a API

Reply:
@vxcamiloxv I would agree with you _except_ that AS2 output is opt-in. So if clients don't want to deal with the hassle of consuming two different formats in input and output (_I_ certainly wouldn't), they don't have to. They can just wait for AS2 input to ship too.

Reply:
Ha ok, yes, probably they will wait for full implementation